deployment/ccip/changeset/solana/cs_deploy_chain.go

```diff
@@ -368,9 +396,44 @@ func deployChainContractsSolana(
 	offRampConfigPDA, _, _ := solState.FindOfframpConfigPDA(offRampAddress)
 	err = chain.GetAccountDataBorshInto(e.GetContext(), offRampConfigPDA, &offRampConfigAccount)
 	if err != nil {
-		if err2 := intializeOffRamp(e, chain, ccipRouterProgram, feeQuoterAddress, offRampAddress, addressLookupTable); err2 != nil {
+		table, err2 := solCommonUtil.SetupLookupTable(
+			e.GetContext(),
+			chain.Client,
+			*chain.DeployerKey,
+			[]solana.PublicKey{
+				// system
+				solana.SystemProgramID,
+				solana.ComputeBudget,
+				solana.SysVarInstructionsPubkey,
+				// token
+				solana.Token2022ProgramID,
+				solana.TokenProgramID,
+				solana.SPLAssociatedTokenAccountProgramID,
+			})
+		if err2 != nil {
+			return fmt.Errorf("failed to create address lookup table: %w", err)
+		}
+		addressLookupTable = table
+		err2 = ab.Save(chain.Selector, addressLookupTable.String(), deployment.NewTypeAndVersion(changeset.OfframpAddressLookupTable, offRampVersion))
+		if err2 != nil {
+			return fmt.Errorf("failed to save address: %w", err)
+		}
+		if err2 := initializeOffRamp(e, chain, ccipRouterProgram, feeQuoterAddress, offRampAddress, addressLookupTable); err2 != nil {
 			return err2
 		}
+		// Initializing a new offramp means we need a new lookup table and need to fully populate it
+		needFQinLookupTable = true
+		needRouterinLookupTable = true
+		offRampConfigPDA, _, _ := solState.FindOfframpConfigPDA(offRampAddress)
+		offRampReferenceAddressesPDA, _, _ := solState.FindOfframpReferenceAddressesPDA(offRampAddress)
+		offRampBillingSignerPDA, _, _ := solState.FindOfframpBillingSignerPDA(offRampAddress)
+		lookupTableKeys = append(lookupTableKeys, []solana.PublicKey{
+			// offramp
+			offRampAddress,
+			offRampConfigPDA,
+			offRampReferenceAddressesPDA,
+			offRampBillingSignerPDA,
+		}...)
```

Comment:
Let's chat a bit next week (not now as this is not urgent) about which other addresses to add here. What I'm saying is just an optimization, it will work as-is, but we can improve and use less tx space on account addresses by having more often-used accounts in the lookup table